clash-prelude/src/Clash/Sized/Internal/Unsigned.hs

```diff
@@ -156,6 +157,7 @@ instance Show (Unsigned n) where
 instance ShowX (Unsigned n) where
   showsPrecX = showsPrecXWith showsPrec
 
+instance NFDataX (Unsigned n) where rnfX = rwhnfX
```

Comment:
Doesn't only evaluate the U constructor?

Reply:
Newtypes don't have a constructor at runtime, so I don't think so!

Reply:
Hmm.. the error GHC throws is confusing:

```
src/Clash/Sized/Internal/Unsigned.hs:139:5-37: error:
    • A newtype constructor cannot have a strictness annotation,
        but ‘U’ does
    • In the definition of data constructor ‘U’
      In the newtype declaration for ‘Unsigned’
    |
139 |     U { unsafeToInteger :: !Integer }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

So it seems to suggest that you can add it to `U`..?

Reply:
Ah nvm, I can't read! So it does seems like the `rnfX = rwhnfX` is correct in this case.

Reply:
Right, with the newtype it makes total sense.